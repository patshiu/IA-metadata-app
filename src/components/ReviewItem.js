import React from 'react';
import PropTypes from 'prop-types';
import ReviewStars from './ReviewStars';

import './ReviewItem.scss';

const ReviewItem = ({ review }) => {
  return (
    <div className="review-item">
      <h2>{review.reviewtitle}</h2>
      <h5>
        <ReviewStars reviewStars={review.stars} /> by {review.reviewer} on {review.reviewdate}
      </h5>
      <p>{review.reviewbody}</p>
    </div>
  );
};

ReviewItem.propTypes = {
  review: PropTypes.shape({
    reviewtitle: PropTypes.string,
    reviewer: PropTypes.string,
    reviewdate: PropTypes.string,
    reviewbody: PropTypes.string,
    stars: PropTypes.string
  }).isRequired
};

export default ReviewItem;
