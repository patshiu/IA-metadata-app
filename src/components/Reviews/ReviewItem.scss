@import '../../_variables';

.review-item {
  border: 1px solid $base-darker;
  border-radius: $border-radius-sm;
  background: $base-lighter;
  padding: 1rem 2rem;
  margin-bottom: 2rem;
}