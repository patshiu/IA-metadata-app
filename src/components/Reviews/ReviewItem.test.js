import React from 'react';
import renderer from 'react-test-renderer';

import ReviewItem from './ReviewItem';

it('renders review card correctly', () => {
  const review = {
    review_id: '6536',
    reviewbody:
      "This short film was written, produced and directed by Charles and Rae Eames for the IBM Pavillion at the 1958 Brussels World's Fair. Animation by John Whitney. Music by Elmer Bernstein. Topic is the computer in the context of human development.  MUST VIEW!",
    reviewtitle: 'A gem from the Late 50s - Charles and Rae Eames',
    reviewer: 'Steve R',
    reviewdate: '2003-10-02 01:25:35',
    createdate: '2003-10-02 01:25:35',
    stars: '5'
  };
  const tree = renderer.create(<ReviewItem review={review} />).toJSON();
  expect(tree).toMatchSnapshot();
});
