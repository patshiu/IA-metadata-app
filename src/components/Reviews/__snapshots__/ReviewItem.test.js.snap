// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders review card correctly 1`] = `
<div
  className="review-item"
>
  <h3>
    A gem from the Late 50s - Charles and Rae Eames
  </h3>
  <div>
    <div
      aria-description="Rating: 5 of 5 stars"
    >
      ★★★★★
    </div>
     by 
    Steve R
     on 
    2003-10-02 01:25:35
  </div>
  <p>
    This short film was written, produced and directed by Charles and Rae Eames for the IBM Pavillion at the 1958 Brussels World's Fair. Animation by John Whitney. Music by Elmer Bernstein. Topic is the computer in the context of human development.  MUST VIEW!
  </p>
</div>
`;
